src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -1518,13 +1518,15 @@ public void Remove(string key)
             IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
         }
 
-        private sealed class TagsLinkedList : IEnumerable<KeyValuePair<string, object?>>
+        internal sealed class TagsLinkedList : IEnumerable<KeyValuePair<string, object?>>
         {
             private DiagNode<KeyValuePair<string, object?>>? _first;
             private DiagNode<KeyValuePair<string, object?>>? _last;
 
             private StringBuilder? _stringBuilder;
 
+            public TagsLinkedList() { }
```

Comment:
>TagsLinkedList [](http://example.com/codeflow?start=19&length=14)

I suggest instead of having this constructor, keep it private and have instead a static property return the empty list. This to prevent anyone misuse TagsLinkedList.

Reply:
I went to make this change but it occurred to me probably not the safest idea to return that static `TagsLinkedList` instance because users _could_ do something screwy to it (like add something reflectively) and put things into a bad state. I removed this ctor and used `Array.Empty` to generate the empty instance instead.